Services/Administration/OJS.Services.Administration.Business/ExamGroups/Permissions/MultipleUsersToExamGroupPermissionService.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace OJS.Services.Administration.Business.ExamGroups.Permissions;
+
+using OJS.Data.Models.Contests;
+using OJS.Services.Administration.Business.Contests;
+using OJS.Services.Administration.Data;
+using OJS.Services.Administration.Models.ExamGroups;
+using OJS.Services.Common.Models.Users;
+using OJS.Services.Infrastructure.Exceptions;
+using System.Threading.Tasks;
+
+public class MultipleUsersToExamGroupPermissionService : IEntityPermissionsService<ExamGroup, MultipleUsersToExamGroupModel>
+{
+    private readonly IContestsBusinessService contestsBusinessService;
+    private readonly IExamGroupsDataService examGroupsDataService;
+
+    public MultipleUsersToExamGroupPermissionService(IContestsBusinessService contestsBusinessService, IExamGroupsDataService examGroupsDataService)
+    {
+        this.contestsBusinessService = contestsBusinessService;
+        this.examGroupsDataService = examGroupsDataService;
+    }
+
+    public async Task<bool> HasPermission(UserInfoModel user, MultipleUsersToExamGroupModel value, string operation)
+    {
+        var contestId = await this.examGroupsDataService.GetContestIdById(value.ExamGroupId);
+        if (!contestId.HasValue)
+        {
+            throw new BusinessServiceException("Unable to add or remove user from exam group which has no contest.");
+        }
```

Comment:
Why not return `false` or just let the `.UserHasContestPermissions` method return it? What is the purpose of an exception here?

Reply:
Because we do not have a logic for permission messages. There is one generic message which says "You are not allowed to perform this operation". With the exception,  better message will be displayed to the user.

Reply:
The button should be disabled with a message like a tooltip or something next to it.

Reply:
Fixed.